helpers/social_connection_helper.py

```diff
@@ -0,0 +1,137 @@
+        return conns_by_type
+
+
+class SocialConnectionParser(object):
+    FACEBOOK_URL_PATTERNS = ['facebook.com/']
+    TWITTER_URL_PATTERNS = ['twitter.com/']
+    YOUTUBE_URL_PATTERNS = ['youtube.com/user/']
+    GITHUB_URL_PATTERNS = ['github.com/']
+
+    @classmethod
+    def partial_social_dict_from_url(cls, url):
+        """
+        Takes a url, and turns it into a partial Media object dict
+        """
+        if any(s in url for s in cls.FACEBOOK_URL_PATTERNS):
+            return cls._partial_social_dict_from_facebook(url)
+        elif any(s in url for s in cls.TWITTER_URL_PATTERNS):
+            return cls._partial_social_dict_from_twitter(url)
+        elif any(s in url for s in cls.YOUTUBE_URL_PATTERNS):
+            return cls._partial_social_dict_from_youtube(url)
+        elif any(s in url for s in cls.GITHUB_URL_PATTERNS):
+            return cls._partial_social_dict_from_github(url)
+        else:
+            logging.warning("Failed to determine media type from url: {}".format(url))
+            return None
+
+    @classmethod
+    def _partial_social_dict_from_facebook(cls, url):
+        social_dict = {}
+        social_dict['social_type_enum'] = SocialConnectionType.FACEBOOK
+        foreign_key = cls._parse_facebook_foreign_key(url)
+        if foreign_key is None:
+            logging.warning("Failed to determine foreign_key from url: {}".format(url))
+            return None
+        social_dict['foreign_key'] = foreign_key
+
+        return social_dict
+
+    @classmethod
+    def _partial_social_dict_from_twitter(cls, url):
+        social_dict = {}
+        social_dict['social_type_enum'] = SocialConnectionType.TWITTER
+        foreign_key = cls._parse_twitter_foreign_key(url)
+        if foreign_key is None:
+            logging.warning("Failed to determine foreign_key from url: {}".format(url))
+            return None
+        social_dict['foreign_key'] = foreign_key
+
+        return social_dict
+
+    @classmethod
+    def _partial_social_dict_from_youtube(cls, url):
+        social_dict = {}
+        social_dict['social_type_enum'] = SocialConnectionType.YOUTUBE
+        foreign_key = cls._parse_youtube_foreign_key(url)
+        if foreign_key is None:
+            logging.warning("Failed to determine foreign_key from url: {}".format(url))
+            return None
+        social_dict['foreign_key'] = foreign_key
+
+        return social_dict
+
+    @classmethod
+    def _partial_social_dict_from_github(cls, url):
+        social_dict = {}
+        social_dict['social_type_enum'] = SocialConnectionType.GITHUB
+        foreign_key = cls._parse_github_foreign_key(url)
+        if foreign_key is None:
+            logging.warning("Failed to determine foreign_key from url: {}".format(url))
+            return None
+        social_dict['foreign_key'] = foreign_key
+
+        return social_dict
+
+    @classmethod
+    def _parse_facebook_foreign_key(cls, url):
```

Comment:
These methods feel repetitive. Is there a way we can abstract them more by passing in a regex?